cpp/cmake/thirdparty/get_cucollections.cmake

```diff
@@ -21,7 +21,7 @@ function(find_and_configure_cucollections)
     cuco 0.0
     GLOBAL_TARGETS cuco::cuco
     CPM_ARGS GITHUB_REPOSITORY NVIDIA/cuCollections
-    GIT_TAG f0eecb203590f1f4ac4a9f1700229f4434ac64dc
+    GIT_TAG 6433e8ad7571f14cc5384051b049029c60dd1ce0
```

Comment:
If the dependency PR is already merged, shouldn't this GIT_TAG be removed and just depend on a release branch?

Reply:
cuco doesn't have a release yet thus we choose to fetch a specific commit instead of the main branch (`dev`).